tests/src/FunctionalJavascript/ModuleIntegrationTest.php

```diff
@@ -542,4 +542,67 @@ protected function liveblogSetBody(DocumentElement $page, $body) {
     );
   }
 
+  /**
+   * Testing integration of "AMP" module and theme.
+   */
+  public function testAmpIntegration() {
+    if (!\Drupal::service('theme_installer')->install(['thunder_amp'])) {
+      $this->fail("thunder_amp theme couldn't be installed.");
+      return;
+    }
+
+    // Create article and save it as unpublished.
+    $this->articleFillNew([
+      'field_channel' => 1,
+      'title[0][value]' => 'AMP Article',
+      'field_seo_title[0][value]' => 'AMP Article',
+    ]);
+
+    // Add all paragraph types.
+    $this->addTextParagraph('field_paragraphs', 'Article Text 1');
+    $this->addImageParagraph('field_paragraphs', ['media:1']);
+    $this->addGalleryParagraph('field_paragraphs', 'AMP Gallery', [
+      'media:5',
+      'media:6',
+      'media:10',
+    ]);
+    $this->addLinkParagraph('field_paragraphs', 'Google', 'https://google.de');
+    $this->addVideoParagraph('field_paragraphs', ['media:7']);
+    $this->addSocialParagraph('field_paragraphs', 'https://www.instagram.com/p/BT8qRIdBz7Q/', 'instagram');
+    $this->addSocialParagraph('field_paragraphs', 'https://twitter.com/ThunderCoreTeam/status/776417570756976640', 'twitter');
+
+    $this->clickArticleSave();
+
+    $this->drupalGet('/node/10', ['query' => ['amp' => 1]]);
+
+    // Text paragraph.
+    $this->assertSession()->pageTextContains('Article Text 1');
+
+    // Image paragraph.
+    $this->assertSession()->elementExists('css', '.paragraph--type--image amp-img');
+    $this->assertSession()->waitForElementVisible('css', '.paragraph--type--image amp-img img');
+
+    // Gallery paragraph.
+    $this->assertSession()->elementExists('css', '.paragraph--type--gallery amp-carousel');
+    // Images in gallery paragraph.
+    $this->assertSession()->elementsCount('css', '.paragraph--type--gallery amp-carousel amp-img', 3);
+    $this->assertSession()->waitForElementVisible('css', '.paragraph--type--gallery amp-carousel amp-img');
```

Comment:
Why is this wait here, since it's already asserted that elements exist? They are not visible in moment of count assertion or?